src/compiler/transformers/esnext.ts

```diff
@@ -16,9 +19,67 @@ namespace ts {
                 return node;
             }
             switch (node.kind) {
+                case SyntaxKind.BinaryExpression:
+                    const binaryExpression = <BinaryExpression>node;
+                    if (isLogicalOrCoalescingAssignmentOperator(binaryExpression.operatorToken.kind)) {
+                        return transformLogicalAssignmentOperators(binaryExpression);
+                    }
+                    // falls through
                 default:
                     return visitEachChild(node, visitor, context);
             }
         }
+
+        function transformLogicalAssignmentOperators(binaryExpression: BinaryExpression): VisitResult<Node> {
+            const operator = binaryExpression.operatorToken;
+            if (isCompoundAssignment(operator.kind) && isLogicalOrCoalescingAssignmentOperator(operator.kind)) {
+                const nonAssignmentOperator = getNonAssignmentOperatorForCompoundAssignment(operator.kind);
+                let left = skipParentheses(visitNode(binaryExpression.left, visitor, isLeftHandSideExpression));
+                let assignmentTarget = left;
+                const right = skipParentheses(visitNode(binaryExpression.right, visitor, isExpression));
+                if (isPropertyAccessExpression(left) || isElementAccessExpression(left)) {

```

Comment:
The `isAccessExpression` function covers both property and element access.
```suggestion
                if (isAccessExpression(left)) {
```